### ISSUE TITLE
Firestore enablePersistence(): “Error enabling offline persistence. Falling back to persistence disabled”

### ISSUE THREAD

**Issue by kebarvid** [firebase/firebase-js-sdk#2404]
### Describe your environment

  * Operating System version: All tested (Win10, OSX El capitain)
  * Browser version: Latest Chrome 6 Firefox
  * Firebase SDK version: 7.5.0
  * Firebase Product: Firestore

### Describe the problem
I'm making a PWA app with Ionic 4 (angular) and Firebase Cloud Firestore as database. And it was working fine until just now when i wanted to enable offline persistance. Using AngularFire and the docs is not hard to follow. This is is how i enable it in my app-module.ts
```typescript
@NgModule({
  declarations: [AppComponent],
  entryComponents: [],
  imports: [
    BrowserModule,
    IonicModule.forRoot(),
    AppRoutingModule,
    ComponentsModule,
    AngularFireModule.initializeApp(environment.firebaseConfig),
    AngularFirestoreModule.enablePersistence(), // <- Trying to enable here
    AngularFireAuthModule,
  ServiceWorkerModule.register('ngsw-worker.js', { enabled: environment.production })
  ],
  providers: [
    { provide: RouteReuseStrategy, useClass: IonicRouteStrategy },
    GlobalService
  ],
  bootstrap: [AppComponent]
})
export class AppModule {}
```
When i run the app i get this warning in Chrome Developer Tools:
> Error enabling offline persistence. Falling back to persistence disabled: TypeError: Cannot read property 'USE_MOCK_PERSISTENCE' of undefined

and a similar one in Firefox:
> Error enabling offline persistence. Falling back to persistence disabled: TypeError: process.env is undefined

#### Steps to reproduce:
enable it in app.module.ts with:
> AngularFirestoreModule.enablePersistence()

#### Relevant Code:

You can se the error live @ https://cubetrainer.nu

**Comment by rommelpe** [comment]
Thanks for reporting this issue, @kebarvid. I was able to encounter the same behavior using your live app. I'm trying also replicate this on my end using my [new fresh app](https://github.com/rommelpe/ionic-angular-angularfire) but wasn't able to replicate it. Can you help me reproduce the error? If you could provide a [minimal repro](https://stackoverflow.com/help/minimal-reproducible-example) of your app that I can run locally, that would be a great help for us to speed up the investigation. Thanks

**Comment by thebrianchen** [comment]
@kebarvid It seems like your app has overwritten the `process` global variable. The error you are getting stems from our check to use mock persistence ([source code](https://github.com/firebase/firebase-js-sdk/blob/master/packages/firestore/src/local/simple_db.ts#L209)), because it seems that `process` is defined, but `process.env` is not. 

I'll add an additional check of `process.env !== undefined` to the check, which should stop the error from appearing, but to successfully enable persistence, you'll have to rename your global `process` to something else.

**Comment by kebarvid** [comment]
You are correct @thebrianchen . To make a database solution i was using earlier to work i did set a global variable called process. When i removed that everything works fine! Thanks for the help!
